layout/rowgroup.go

```diff
@@ -68,20 +67,6 @@ func ReadRowGroup(rowGroupHeader *parquet.RowGroup, PFile source.ParquetFile, sc
 
 		wg.Add(1)
 		go func(index int64, bgn int64, end int64) {
-			defer func() {
-				wg.Done()
```

Comment:
You need to leave a `defer wg.Done()` here, otherwise the method will block indefinitely when waiting.

Reply:
You are absolutely right, I will update this PR once I get onto my personal laptop.